modules/internal/ChapelIO.chpl

```diff
@@ -465,7 +465,7 @@ module ChapelIO {
         if st == QIO_AGGREGATE_FORMAT_JSON {
           start = new ioLiteral("{");
         } else if st == QIO_AGGREGATE_FORMAT_CHPL {
-          start = new ioLiteral("new " + typeToString(t) + "(");
+          start = new ioLiteral("new " + typeToString(t).c_str() + "(");
```

Comment:
It looks like we support both ': c_string" and ".c_str()" on param strings, is that right?  What guidance would suggest using one or the other?